In [445]:
import pandas as pd
import numpy as np

# 2015-2016 Data

In [446]:
# 2015-2016
path = '/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/2015-2016'

import os
os.chdir(path)

## PHQ-9 Survey - part of Questionnaire

In [447]:
DPQ_2015_2016 = pd.read_csv('./Questionnaire/DPQ_I.csv')

In [448]:
DPQ_2015_2016.head()

SEQN        DPQ010        DPQ020        DPQ030        DPQ040  \
0  83732.0  5.397605e-79  5.397605e-79  5.397605e-79  1.000000e+00   
1  83733.0  1.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79   
2  83734.0  5.397605e-79  5.397605e-79  5.397605e-79  1.000000e+00   
3  83735.0  1.000000e+00  1.000000e+00  2.000000e+00  2.000000e+00   
4  83736.0  1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   

         DPQ050        DPQ060        DPQ070        DPQ080        DPQ090  \
0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
1  1.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  1.000000e+00  3.000000e+00  2.000000e+00  5.397605e-79  1.000000e+00   
4  3.000000e+00  5.397605e-79  1.000000e+00  5.397605e-79  5.397605e-79   

         DPQ100  
0  5.397605e-79  
1  5.397605e-79  
2  1.000000e+00  
3  5.397605e-79  
4  5.397605e-79

In [449]:
# drop final follow-up question
DPQ_2015_2016.drop(['DPQ100'], axis=1, inplace=True)

In [450]:
# values less than 1 are zero
def set_zero(df):
    df[df<1] = 0
    
set_zero(DPQ_2015_2016)

In [451]:
# need to drop rows where not all col are complete
DPQ_2015_2016.dropna(axis=0, how='any', thresh=None, subset=None, inplace=True)

In [452]:
# take a sum across rows and save as new col
DPQ_2015_2016['sum'] = DPQ_2015_2016.iloc[:,1:].sum(axis=1)

In [453]:
# create new target label 'depressed' if sum is greater than 10 --> 1, otherwise 0
DPQ_2015_2016.loc[DPQ_2015_2016['sum'] > 10, 'depressed'] = 1
DPQ_2015_2016.loc[DPQ_2015_2016['sum'] <= 10, 'depressed'] = 0

In [454]:
DPQ_2015_2016.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0  83732.0     0.0     0.0     0.0     1.0     0.0     0.0     0.0     0.0   
1  83733.0     1.0     0.0     0.0     0.0     1.0     0.0     0.0     0.0   
2  83734.0     0.0     0.0     0.0     1.0     0.0     0.0     0.0     0.0   
3  83735.0     1.0     1.0     2.0     2.0     1.0     3.0     2.0     0.0   
4  83736.0     1.0     1.0     1.0     1.0     3.0     0.0     1.0     0.0   

   DPQ090   sum  depressed  
0     0.0   1.0        0.0  
1     0.0   2.0        0.0  
2     0.0   1.0        0.0  
3     1.0  13.0        1.0  
4     0.0   8.0        0.0

In [455]:
DPQ_2015_2016.shape

(5161, 12)

In [456]:
DPQ_I = DPQ_2015_2016[['SEQN', 'depressed']]

In [457]:
DPQ_I.head()

SEQN  depressed
0  83732.0        0.0
1  83733.0        0.0
2  83734.0        0.0
3  83735.0        1.0
4  83736.0        0.0

In [458]:
len(DPQ_2015_2016[DPQ_2015_2016['depressed']==1])  # about 8% depressed = true

352

## Demographic Data

### Demographic Variables and Sample Weights (DEMO_I)

In [459]:
# start with demographic data
DEMO_2015_2016 = pd.read_csv('./Demographics/DEMO_I.csv')

In [460]:
DEMO_2015_2016.head(10)

SEQN  SDDSRVYR  RIDSTATR  RIAGENDR  RIDAGEYR  RIDAGEMN  RIDRETH1  \
0  83732.0       9.0       2.0       1.0      62.0       NaN       3.0   
1  83733.0       9.0       2.0       1.0      53.0       NaN       3.0   
2  83734.0       9.0       2.0       1.0      78.0       NaN       3.0   
3  83735.0       9.0       2.0       2.0      56.0       NaN       3.0   
4  83736.0       9.0       2.0       2.0      42.0       NaN       4.0   
5  83737.0       9.0       2.0       2.0      72.0       NaN       1.0   
6  83738.0       9.0       2.0       2.0      11.0       NaN       1.0   
7  83739.0       9.0       2.0       1.0       4.0       NaN       3.0   
8  83740.0       9.0       2.0       1.0       1.0      13.0       2.0   
9  83741.0       9.0       2.0       1.0      22.0       NaN       4.0   

   RIDRETH3  RIDEXMON  RIDEXAGM  ...  DMDHREDU  DMDHRMAR  DMDHSEDU  \
0       3.0       1.0       NaN  ...       5.0       1.0       3.0   
1       3.0       1.0       NaN  ...       3.0       3.0       NaN   
2       3.0       2.0       NaN  ...       3.0       1.0       3.0   
3       3.0       2.0       NaN  ...       5.0       6.0       NaN   
4       4.0       2.0       NaN  ...       4.0       3.0       NaN   
5       1.0       1.0       NaN  ...       5.0       1.0       5.0   
6       1.0       1.0     141.0  ...       1.0       2.0       NaN   
7       3.0       2.0      54.0  ...       5.0       1.0       5.0   
8       2.0       2.0      14.0  ...       9.0       5.0       NaN   
9       4.0       1.0       NaN  ...       4.0       1.0       5.0   

        WTINT2YR       WTMEC2YR  SDMVPSU  SDMVSTRA  INDHHIN2  INDFMIN2  \
0  134671.370419  135629.507405      1.0     125.0      10.0      10.0   
1   24328.560239   25282.425927      1.0     125.0       4.0       4.0   
2   12400.008522   12575.838818      1.0     131.0       5.0       5.0   
3  102717.995647  102078.634508      1.0     131.0      10.0      10.0   
4   17627.674984   18234.736219      2.0     126.0       7.0       7.0   
5   11252.311968   10878.684022      1.0     128.0      14.0      14.0   
6    9964.725426    9860.624532      1.0     120.0       6.0       6.0   
7   44749.889547   46173.307234      2.0     124.0      15.0      15.0   
8    9891.944044   10963.313710      1.0     119.0      77.0      77.0   
9   37043.087007   39353.307397      2.0     128.0       7.0       7.0   

   INDFMPIR  
0      4.39  
1      1.32  
2      1.51  
3      5.00  
4      1.23  
5      2.82  
6      1.18  
7      4.22  
8       NaN  
9      2.08  

[10 rows x 47 columns]

Demographics variables:
- DMDBORN4: Due to the concerns of disclosure risk, starting in 2011, country of birth was recoded into two categories: 1) Born in 50 U.S. states or Washington, DC; and 2) Born in other countries, including U.S. territories.

- DMDEDUC2: This variable is the highest grade or level of education completed by adults 20 years and older. The response categories are: less than 9th grade education, 9-11th grade education (includes 12th grade and no diploma), High school graduate/GED, some college or associates (AA) degree, and college graduate or higher.

- DMDHHSIZ: This variable is the number of people in the participant’s household. The values for this variable range from 1 to 7. Due to disclosure concerns, households that are comprised of 7 or more people are included in the category that is labeled ‘7 or more’.

- RIDAGEYR: Age in years, at the time of the screening interview, is reported for survey participants between the ages of 1 and 79 years of age. All responses of participants aged 80 years and older are coded as ‘80.’

- RIAGENDR: Gender: male 1, female 2

- INDHHIN2: Total household income (reported as a range value in dollars), range 1-15, 77 refused, 99 dont know

- DMQMIL1Z: Served active duty in US Armed Forces, 1 yes, 0 no, 7 refuse, 9 dont know

- DMDMARTL: Marital status - 1 Married, 2 widowed, 3 divorced, 4 separated, 5 never married, 6 living with partner, 77 refused	


In [461]:
# filter only relevant cols
DEMO_I = DEMO_2015_2016.filter(['SEQN','RIDRETH1', 'DMDEDUC2', 'DMDHHSIZ', 'RIDAGEYR', 'RIAGENDR','INDHHIN2', 
                        'DMQMIL1Z', 'DMDMARTL'], axis=1)

In [462]:
# rename col to be more useful
demo_cols = {'SEQN': 'SEQN', 'RIDRETH1':'race_ethnicity', 'DMDEDUC2': 'edu_level', 'DMDHHSIZ': '#_ppl_household', 
               'RIDAGEYR': 'age', 'RIAGENDR': 'gender','INDHHIN2': 'annual_HI', 
                        'DMQMIL1Z': 'served_army', 'DMDMARTL':'marital_status'}

DEMO_I.columns = DEMO_I.columns.map(demo_cols)

In [463]:
DEMO_I.head()

SEQN  race_ethnicity  edu_level  #_ppl_household   age  gender  \
0  83732.0             3.0        5.0              2.0  62.0     1.0   
1  83733.0             3.0        3.0              1.0  53.0     1.0   
2  83734.0             3.0        3.0              2.0  78.0     1.0   
3  83735.0             3.0        5.0              1.0  56.0     2.0   
4  83736.0             4.0        4.0              5.0  42.0     2.0   

   annual_HI  marital_status  
0       10.0             1.0  
1        4.0             3.0  
2        5.0             1.0  
3       10.0             6.0  
4        7.0             3.0

## Dietary Data

### Dietary Interview - Total Nutrient Intakes, First Day (DR1TOT_I)

In [464]:
DIET_2015_2016 = pd.read_csv('./Dietary/DR1TOT_I.csv')

Dietary variables:
- DR1TCAFF : caffeine - Caffeine (mg) range 0 to 2448
- DRQSDIET - On special diet? - 1,2,9, missing
- DR1TPROT - Protein (gm) - 0 to 869.49, missing
- DR1TCARB - Carbohydrate (gm) - 8.67 to 1423.87, missing
- DR1TSUGR - Total sugars (gm) - 0.13 to 1115.5, missing
- DR1TFIBE - Dietary fiber (gm) - 0 to 136.6, missing
- DR1TTFAT - Total fat (gm) - 0 to 548.38, missing
- DR1TCHOL - Cholesterol (mg) - 0 to 3515, missing
- DR1TVB6 - Vitamin B6 (mg)
- DR1TFOLA - Total folate (mcg) - 0 to 3172, missing
- DR1TVB12 - Vitamin B12 (mcg) - 0 to 110, missing
- DR1TVD - Vitamin D (D2 + D3) (mcg) - 0 to 86.3, missing
- DR1TMAGN - Magnesium (mg) - 0 2725, missing
- DR1TIRON - Iron (mg) - 0 to 130, missing
- DR1TZINC - Zinc (mg) - 0 to 116, missing
- DR1TSELE - Selenium (mcg) - 0 to 986.5, missing
- DR1_320Z - Total plain water drank yesterday (gm) - 0 13440, missing

In [465]:
# filter only relevant cols
DIET_I = DIET_2015_2016.filter(['SEQN', 'DR1TCAFF', 'DRQSDIET', 'DR1TPROT','DR1TCARB','DR1TSUGR','DR1TFIBE',
                               'DR1TTFAT','DR1TCHOL','DR1TVB6','DR1TFOLA','DR1TVB12','DR1TVD',
                               'DR1TMAGN','DR1TIRON','DR1TZINC','DR1TSELE','DR1_320Z'], axis=1)

diet_cols = {'SEQN': 'SEQN', 'DR1TCAFF':'caffeine','DRQSDIET': 'special_diet', 'DR1TPROT':'protein',
             'DR1TCARB':'carb','DR1TSUGR':'sugar','DR1TFIBE':'fiber',
            'DR1TTFAT':'fat','DR1TCHOL':'diet_chol','DR1TVB6':'VitB6','DR1TFOLA':'folate',
             'DR1TVB12':'VitB12','DR1TVD':'VitD', 'DR1TMAGN':'magnesium',
             'DR1TIRON':'iron','DR1TZINC':'zinc','DR1TSELE':'selenium','DR1_320Z':'water'}

DIET_I.columns = DIET_I.columns.map(diet_cols)

In [466]:
DIET_I.head()

SEQN      caffeine  special_diet  protein    carb   sugar  fiber    fat  \
0  83732.0  3.600000e+02           2.0    76.03  193.29   42.31   23.6  79.24   
1  83733.0  1.920000e+02           2.0    62.36  356.85  180.84    7.3  77.91   
2  83734.0  3.060000e+02           1.0   116.50  304.53   62.87   35.1  91.97   
3  83735.0  2.480000e+02           2.0    70.38  157.99   54.77   14.0  49.23   
4  83736.0  5.397605e-79           2.0    19.30   90.30   71.84    2.0  19.63   

   diet_chol  VitB6  folate  VitB12          VitD  magnesium   iron   zinc  \
0      138.0  2.279   450.0    1.89  2.600000e+00      255.0  16.01   8.73   
1      407.0  1.823   478.0    2.92  4.600000e+00      262.0  11.01   4.82   
2      506.0  3.117   553.0    2.35  4.700000e+00      497.0  26.17  14.09   
3      138.0  1.259   175.0    3.08  6.900000e+00      318.0   9.07   7.59   
4       96.0  0.409    52.0    0.36  5.397605e-79       42.0   1.83   1.54   

   selenium         water  
0     113.6  9.600000e+02  
1      89.9  5.070000e+02  
2     140.9  8.700000e+02  
3     104.5  1.800000e+02  
4      27.7  5.397605e-79

In [467]:
len(DIET_I.columns)

18

## Questionnaire Data

### Diabetes

variables
- DIQ010 : doc_diabetes - {Have you/Has SP}} ever been told by a doctor or health professional that {you have/{he/she/SP} has} diabetes or sugar diabetes? 1 yes, 2 no, 3 boarderline
- DIQ160 : doc_prediabetes - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/SP has} any of the following: prediabetes, impaired fasting glucose, impaired glucose tolerance, borderline diabetes or that {your/her/his} blood sugar is higher than normal but not high enough to be called diabetes or sugar diabetes? 1 yes, 2 no

In [468]:
DIQ_2015_2016 = pd.read_csv('./Questionnaire/DIQ_I.csv')

In [469]:
DIQ_2015_2016.columns

Index(['SEQN', 'DIQ010', 'DID040', 'DIQ160', 'DIQ170', 'DIQ172', 'DIQ175A',
       'DIQ175B', 'DIQ175C', 'DIQ175D', 'DIQ175E', 'DIQ175F', 'DIQ175G',
       'DIQ175H', 'DIQ175I', 'DIQ175J', 'DIQ175K', 'DIQ175L', 'DIQ175M',
       'DIQ175N', 'DIQ175O', 'DIQ175P', 'DIQ175Q', 'DIQ175R', 'DIQ175S',
       'DIQ175T', 'DIQ175U', 'DIQ175V', 'DIQ175W', 'DIQ175X', 'DIQ180',
       'DIQ050', 'DID060', 'DIQ060U', 'DIQ070', 'DIQ230', 'DIQ240', 'DID250',
       'DID260', 'DIQ260U', 'DIQ275', 'DIQ280', 'DIQ291', 'DIQ300S', 'DIQ300D',
       'DID310S', 'DID310D', 'DID320', 'DID330', 'DID341', 'DID350', 'DIQ350U',
       'DIQ360', 'DIQ080'],
      dtype='object')

In [470]:
DIQ_I = DIQ_2015_2016.filter(['SEQN','DIQ010', 'DIQ160'], axis=1)

In [471]:
# rename col to be more useful
DIQ_cols = {'SEQN':'SEQN', 'DIQ010':'doc_diabetes', 'DIQ160': 'doc_prediabetes'}

DIQ_I.columns = DIQ_I.columns.map(DIQ_cols)

In [472]:
DIQ_I.head()

SEQN  doc_diabetes  doc_prediabetes
0  83732.0           1.0              NaN
1  83733.0           2.0              2.0
2  83734.0           1.0              NaN
3  83735.0           2.0              1.0
4  83736.0           2.0              2.0

### Diet Questionnaire

Variables:
- CBQ505 : fast_food_12m - In the past 12 months, did you buy food from fast food or pizza places?
- DBD895 : #meals_not_Iome - During the past 7 days, how many meals {did you/did SP} get that were prepared away from home in places such as restaurants, fast food places, food stands, grocery stores, or from vending machines?
    - 0 None, 1 to 21 range, 5555 more than 21 meals per week, 7777 refused, 9999 dont know
- DBD900 : #meals_fast_food - How many of those meals {did you/did SP} get from a fast-food or pizza place?
- DBQ700 : how_Iealthy_diet - In general, how healthy is {your/his/her} overall diet? 
    - Excellent to Poor: 1-5

In [473]:
DBQ_2015_2016 = pd.read_csv('./Questionnaire/DBQ_I.csv')

In [474]:
DBQ_2015_2016.columns

Index(['SEQN', 'DBQ010', 'DBD030', 'DBD041', 'DBD050', 'DBD055', 'DBD061',
       'DBQ073A', 'DBQ073B', 'DBQ073C', 'DBQ073D', 'DBQ073E', 'DBQ073U',
       'DBQ700', 'DBQ197', 'DBQ223A', 'DBQ223B', 'DBQ223C', 'DBQ223D',
       'DBQ223E', 'DBQ223U', 'DBQ229', 'DBQ235A', 'DBQ235B', 'DBQ235C',
       'DBQ301', 'DBQ330', 'DBQ360', 'DBQ370', 'DBD381', 'DBQ390', 'DBQ400',
       'DBD411', 'DBQ421', 'DBQ424', 'DBD895', 'DBD900', 'DBD905', 'DBD910',
       'CBQ596', 'CBQ606', 'CBQ611', 'CBQ505', 'CBQ535', 'CBQ540', 'CBQ545',
       'CBQ550', 'CBQ552', 'CBQ580', 'CBQ585', 'CBQ590'],
      dtype='object')

In [475]:
DBQ_I = DBQ_2015_2016.filter(['SEQN','DBD900', 'DBQ700'], axis=1)

In [476]:
# rename col to be more useful
DBQ_cols = {'SEQN':'SEQN',
                  'DBD900':'#meals_fast_food', 'DBQ700': 'how_healthy_diet'}

DBQ_I.columns = DBQ_I.columns.map(DBQ_cols)

In [477]:
DBQ_I.head()

SEQN  #meals_fast_food  how_healthy_diet
0  83732.0      5.397605e-79               3.0
1  83733.0               NaN               1.0
2  83734.0      4.000000e+00               4.0
3  83735.0      1.000000e+00               4.0
4  83736.0      1.000000e+00               5.0

### Sight & Hearing

variables:
- DLQ010 : difficult_Iearing - 1 yes, 2 no
- DLQ020 : difficult_seeing - 1 yes, 2 no
- DLQ050 - Have serious difficulty walking ? 
    - 1=yes, 2=no, 7/9
- DLQ080 - Have difficulty doing errands alone ? 
    - 1=yes, 2=no, 7/9


In [478]:
DLQ_2015_2016 = pd.read_csv('./Questionnaire/DLQ_I.csv')

In [479]:
DLQ_2015_2016.columns

Index(['SEQN', 'DLQ010', 'DLQ020', 'DLQ040', 'DLQ050', 'DLQ060', 'DLQ080',
       'DLQ100', 'DLQ110', 'DLQ130', 'DLQ140', 'DLQ150', 'DLQ170'],
      dtype='object')

In [480]:
DLQ_I = DLQ_2015_2016.filter(['SEQN','DLQ010','DLQ020','DLQ050','DLQ080'], axis=1)

In [481]:
# rename col to be more useful
DLQ_cols = {'SEQN':'SEQN', 'DLQ010': 'difficult_hearing', 'DLQ020': 'difficult_seeing',
           'DLQ050':'difficult_walking','DLQ080':'difficult_errands'}

DLQ_I.columns = DLQ_I.columns.map(DLQ_cols)

In [482]:
DLQ_I.head()

SEQN  difficult_hearing  difficult_seeing  difficult_walking  \
0  83732.0                2.0               2.0                2.0   
1  83733.0                2.0               2.0                2.0   
2  83734.0                1.0               2.0                2.0   
3  83735.0                2.0               2.0                2.0   
4  83736.0                2.0               2.0                1.0   

   difficult_errands  
0                2.0  
1                2.0  
2                2.0  
3                2.0  
4                2.0

### Drugs

variables: 
- DUQ211 : used_marijuana - Have you ever smoked marijuana or hashish at least once a month for more than one year?
- DUQ230 : # days used marijuana or hashish/month
- DUQ240 : used_CMH - Have you ever used cocaine, crack cocaine, heroin, or methamphetamine?
- DUQ370 : injected_drugs - Have you ever, even once, used a needle to inject a drug not prescribed by a doctor?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [483]:
DUQ_2015_2016 = pd.read_csv('./Questionnaire/DUQ_I.csv')

In [484]:
DUQ_I = DUQ_2015_2016.filter(['SEQN','DUQ240'], axis=1)

In [485]:
DUQ_cols = {'SEQN':'SEQN', 'DUQ240':'used_CMH'}

DUQ_I.columns = DUQ_I.columns.map(DUQ_cols)

In [486]:
DUQ_I.head()

SEQN  used_CMH
0  83732.0       1.0
1  83733.0       2.0
2  83735.0       2.0
3  83736.0       2.0
4  83741.0       2.0

### Health Care

HSQ_I
- HSD010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know

HUQ_I
- HUQ010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know
- HUQ041 - Type place most often go for healthcare
1- Clinic or health center
2 - Doctor's office or HMO
3 - Hospital emergency room
4 - Hospital outpatient department	
5 - Some other place
6 - Doesn't go to one place most often
77 - Refused, 99 - Don't know
- HUQ051 - #times receive healthcare over past year
0 - 8 are ranges, 77, 99
- HUQ090 - Seen mental health professional/past yr
1, 2, 7, 9


In [487]:
HUQ_2015_2016 = pd.read_csv('./Questionnaire/HUQ_I.csv')

HUQ_I = HUQ_2015_2016.filter(['SEQN', 'HUQ010', 'HUQ041','HUQ051'], axis=1)
HUQ_cols = {'SEQN':'SEQN', 'HUQ010': 'gen_health_cond','HUQ041':'place_healthcare',
            'HUQ051':'healthcare_past_yr'}

HUQ_I.columns = HUQ_I.columns.map(HUQ_cols)

In [488]:
HUQ_I.head()

SEQN  gen_health_cond  place_healthcare  healthcare_past_yr
0  83732.0              3.0               1.0        5.000000e+00
1  83733.0              2.0               6.0        5.397605e-79
2  83734.0              4.0               5.0        2.000000e+00
3  83735.0              2.0               2.0        4.000000e+00
4  83736.0              5.0               1.0        2.000000e+00

### Health Insurance

variables:
- HIQ011 : health_insurance - {Are you/Is SP} covered by health insurance or some other kind of health care plan? 1 yes, 2 no

In [489]:
HIQ_2015_2016 = pd.read_csv('./Questionnaire/HIQ_I.csv')

HIQ_I = HIQ_2015_2016.filter(['SEQN', 'HIQ011'], axis=1)
HIQ_cols = {'SEQN':'SEQN', 'HIQ011': 'health_insurance'}

HIQ_I.columns = HIQ_I.columns.map(HIQ_cols)

In [490]:
HIQ_I.head()

SEQN  health_insurance
0  83732.0               1.0
1  83733.0               2.0
2  83734.0               1.0
3  83735.0               1.0
4  83736.0               1.0

### Househole Characteristics

HOQ_I
- HOD050 - Number of rooms in home
1-13, 777, 999
- HOQ065 - Home owned, bought, rented, other
1 - Owned or being bought
2 - Rented
3 - Other arrangements
7 - Refused, 9 - Don't know


In [491]:
HOQ_2015_2016 = pd.read_csv('./Questionnaire/HOQ_I.csv')

HOQ_I = HOQ_2015_2016.filter(['SEQN', 'HOD050','HOQ065'], axis=1)
HOQ_cols = {'SEQN':'SEQN', 'HOD050': '#_rooms','HOQ065':'home_ownership'}

HOQ_I.columns = HOQ_I.columns.map(HOQ_cols)

In [492]:
HOQ_I.head()

SEQN  #_rooms  home_ownership
0  83732.0      7.0             1.0
1  83733.0      6.0             1.0
2  83734.0      5.0             1.0
3  83735.0      4.0             1.0
4  83736.0      5.0             2.0

### Early Childhood

ECQ_I
- ECQ020 - Mother smoked when pregnant
1=yes, 2=no, 7/9

In [493]:
ECQ_2015_2016 = pd.read_csv('./Questionnaire/ECQ_I.csv')

ECQ_I = ECQ_2015_2016.filter(['SEQN', 'ECQ020'], axis=1)
ECQ_cols = {'SEQN':'SEQN', 'ECQ020': 'mom_smoked_preg'}

ECQ_I.columns = ECQ_I.columns.map(ECQ_cols)

In [494]:
ECQ_I.head()

SEQN  mom_smoked_preg
0  83738.0              1.0
1  83739.0              2.0
2  83740.0              2.0
3  83745.0              1.0
4  83746.0              2.0

### Food Security

FSQ_I
- FSD032A - HH Worried run out of food
    - 1 - Often true
    - 2 - Sometimes true
    -  - Never true
    - 7 - Refused, 9 - Don't know
- FSD032B - HH Food didn't last
    - “ “
- FSD032C - HH Couldn't afford balanced meals
    - “ “
- FSD041 - HH Adults cut size or skip meals
    - 1, 2, 7, 9
- FSD032D - HH Relied on low-cost food for child
1 - Often true
2 - Sometimes true
3 - Never true
7 - Refused, 9 - Don't know
- FSDHH - Household food security category
1 - HH full food security: 0
2 - HH marginal food security: 1-2
3 - HH low food security: 3-5 (HH w/o child) / 3-7 (HH w/ child)
4 - HH very low food security: 6-10 (HH w/o child) / 8-18 (HH w/ child)
No 7/9, just missing
- FSQ012 - HH FS benefit: receive in last 12 months
1, 2, 7, 9
- FSQ162 - HH WIC benefit: receive in last 12 month
1,2,7,9


In [495]:
FSQ_2015_2016 = pd.read_csv('./Questionnaire/FSQ_I.csv')

FSQ_I = FSQ_2015_2016.filter(['SEQN', 'FSD032A', 'FSD032B', 'FSD032C', 'FSD041', 'FSD032D', 'FSDHH',
                             'FSQ012', 'FSQ162'], axis=1)

 
FSQ_cols = {'SEQN': 'SEQN', 'FSD032A':'run_out_food', 'FSD032B':'food_didnt_last', 'FSD032C':'afford_balanced_meals', 
             'FSD041':'adults_skip_meals', 'FSD032D':'relied_cheap_food', 'FSDHH':'food_security_cat',
             'FSQ012':'food_stamps', 'FSQ162':'WIC'}

FSQ_I.columns = FSQ_I.columns.map(FSQ_cols)

In [496]:
FSQ_I.head()

SEQN  run_out_food  food_didnt_last  afford_balanced_meals  \
0  83732.0           3.0              3.0                    3.0   
1  83733.0           3.0              3.0                    3.0   
2  83734.0           2.0              3.0                    3.0   
3  83735.0           3.0              3.0                    3.0   
4  83736.0           2.0              2.0                    3.0   

   adults_skip_meals  relied_cheap_food  food_security_cat  food_stamps  WIC  
0                NaN                NaN                1.0          NaN  2.0  
1                NaN                NaN                1.0          2.0  NaN  
2                2.0                NaN                2.0          2.0  NaN  
3                NaN                NaN                1.0          NaN  2.0  
4                2.0                2.0                3.0          1.0  2.0

In [497]:
len(FSQ_I.columns)

9

### Income

INQ_I
- IND235 - Monthly family income
    - 1-12 ranges, 77, 99
- INQ244 - Family has savings more than $5000
    - 1,2,7,9


In [498]:
INQ_2015_2016 = pd.read_csv('./Questionnaire/INQ_I.csv')

INQ_I = INQ_2015_2016.filter(['SEQN', 'IND235', 'INQ244'], axis=1)

INQ_cols = {'SEQN': 'SEQN', 'IND235':'monthly_income', 'INQ244':'savings_5000'}

INQ_I.columns = INQ_I.columns.map(INQ_cols)


In [499]:
INQ_I.head()

SEQN  monthly_income
0  83732.0            10.0
1  83733.0             1.0
2  83734.0             6.0
3  83735.0             7.0
4  83736.0             7.0

### Medical Conditions

variables:
- MCQ010 : doc_asthma - Has a doctor or other health professional ever told {you/SP} that {you have/s/he/SP has} asthma (az-ma)?
- MCQ080 : doc_overweight - Has a doctor or other health professional ever told {you/SP} that {you were/s/he/SP was} overweight?
- MCQ160a :  doc_arthritis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had arthritis (ar-thry-tis)?
- MCQ160b :  doc_CHF - Has a doctor or other health professional ever told {you/SP} that {you/s/he}had congestive heart failure?
- MCQ160c : doc_CHD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had coronary (kor-o-nare-ee) heart disease?
- MCQ160e : doc_Ieart_attack - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had a heart attack (also called myocardial infarction (my-o-car-dee-al in-fark-shun))?
- MCQ160f : doc_stroke - Has a doctor or other health professional ever told {you/SP} that {you/s/her} had a stroke?
- MCQ160k : doc_chronic_bronchitis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had chronic bronchitis?
- MCQ160l : doc_liver_condition - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had any kind of liver condition?
- MCQ160m : doc_thyroid_problem - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had another thyroid (thigh-roid) problem?
- MCQ160o : doc_COPD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had COPD?
- MCQ220 : doc_cancer - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had cancer or a malignancy (ma-lig-nan-see) of any kind?
- MCQ365a : doc_lose_weight - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: control {your/his/her} weight or lose weight?
- MCQ365b : doc_exercise - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: increase {your/his/her} physical activity or exercise?
- MCQ053 - Taking treatment for anemia/past 3 mos
    - 1,2,7,9
- MCQ084 - Difficulties in thinking or remembering
    - 1, 2, 7, 9
- MCQ230a - What kind of cancer
    - 10-39 different kinds, 66 = more than 3, 77, 99


In [500]:
MCQ_2015_2016 = pd.read_csv('./Questionnaire/MCQ_I.csv')

MCQ_I = MCQ_2015_2016.filter(['SEQN', 'MCQ010', 'MCQ080', 'MCQ160A', 'MCQ160B', 'MCQ160C', 'MCQ160E',
 'MCQ160F', 'MCQ160K', 'MCQ160L', 'MCQ160M', 'MCQ160O', 'MCQ220','MCQ053',
                             'MCQ084','MCQ230A'], axis=1)

 
MCQ_cols = {'SEQN': 'SEQN', 'MCQ010':'doc_asthma', 'MCQ080':'doc_overweight', 'MCQ160A':'doc_arthritis', 
 'MCQ160B':'doc_CHF', 'MCQ160C':'doc_CHD', 'MCQ160E':'doc_heart_attack',
 'MCQ160F':'doc_stroke', 'MCQ160K':'doc_chronic_bronchitis', 'MCQ160L':'doc_liver_condition', 
 'MCQ160M':'doc_thyroid_problem', 'MCQ160O':'doc_COPD', 'MCQ220':'doc_cancer','MCQ053': 'anemia',
                             'MCQ084':'difficult_remembering','MCQ230A':'cancer_type'}

MCQ_I.columns = MCQ_I.columns.map(MCQ_cols)

In [501]:
MCQ_I.head()

SEQN  doc_asthma  doc_overweight  doc_arthritis  doc_CHF  doc_CHD  \
0  83732.0         2.0             1.0            1.0      2.0      2.0   
1  83733.0         2.0             2.0            2.0      2.0      2.0   
2  83734.0         1.0             1.0            1.0      2.0      2.0   
3  83735.0         2.0             1.0            2.0      2.0      2.0   
4  83736.0         1.0             2.0            1.0      2.0      2.0   

   doc_heart_attack  doc_stroke  doc_chronic_bronchitis  doc_liver_condition  \
0               2.0         2.0                     2.0                  2.0   
1               2.0         2.0                     2.0                  2.0   
2               1.0         2.0                     2.0                  1.0   
3               2.0         2.0                     2.0                  2.0   
4               2.0         2.0                     2.0                  2.0   

   doc_thyroid_problem  doc_COPD  doc_cancer  anemia  cancer_type  
0                  2.0       2.0         1.0     2.0         25.0  
1                  2.0       2.0         2.0     2.0          NaN  
2                  1.0       1.0         1.0     2.0         33.0  
3                  2.0       2.0         2.0     2.0          NaN  
4                  2.0       2.0         2.0     2.0          NaN

In [502]:
len(MCQ_I.columns)

15

### Kidney Conditions - Urology (KIQ_U_I)

variables
- KIQ022: doc_kidney - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had weak or failing kidneys? Do not include kidney stones, bladder infections, or incontinence.

In [503]:
KIQ_U_2015_2016 = pd.read_csv('./Questionnaire/KIQ_U_I.csv')

KIQ_U_I = KIQ_U_2015_2016.filter(['SEQN', 'KIQ022'], axis=1)
KIQ_U_cols = {'SEQN':'SEQN', 'KIQ022': 'doc_kidney'}

KIQ_U_I.columns = KIQ_U_I.columns.map(KIQ_U_cols)

In [504]:
KIQ_U_I.head()

SEQN  doc_kidney
0  83732.0         2.0
1  83733.0         2.0
2  83734.0         1.0
3  83735.0         2.0
4  83736.0         2.0

### Occupation

OCQ_I
- OCD150 - Type of work done last week
1 - Working at a job or business
2 - With a job or business but not at work
3 - Looking for work, or
4 - Not working at a job or business?
7 - Refused, 9-  Don’t know
- OCQ180 - Hours worked last week at all jobs
    - Range 0-120 hours
    - 77777, 999999


In [505]:
OCQ_2015_2016 = pd.read_csv('./Questionnaire/OCQ_I.csv')

OCQ_I = OCQ_2015_2016.filter(['SEQN', 'OCD150','OCQ180'], axis=1)
OCQ_cols = {'SEQN':'SEQN', 'OCD150': 'work_type','OCQ180':'hrs_worked_week'}

OCQ_I.columns = OCQ_I.columns.map(OCQ_cols)

In [506]:
OCQ_I.head()

SEQN  work_type  hrs_worked_week
0  83732.0        1.0             60.0
1  83733.0        4.0              NaN
2  83734.0        4.0              NaN
3  83735.0        2.0              NaN
4  83736.0        1.0             32.0

### Oral Health

OHQ_I
- OHQ680 - Last yr embarrassed because of mouth
    - 1 very often to 5 never, 7, 9


In [507]:
OHQ_2015_2016 = pd.read_csv('./Questionnaire/OHQ_I.csv')

OHQ_I = OHQ_2015_2016.filter(['SEQN', 'OHQ680'], axis=1)
OHQ_cols = {'SEQN':'SEQN', 'OHQ680': 'embarassed_mouth'}

OHQ_I.columns = OHQ_I.columns.map(OHQ_cols)

In [508]:
OHQ_I.head()

SEQN  embarassed_mouth
0  83732.0               4.0
1  83733.0               5.0
2  83734.0               5.0
3  83735.0               5.0
4  83736.0               5.0

### Osteoperosis

variables:
- OSQ010A : broken_Iip - Has a doctor ever told {you/SP} that {you/s/he} had broken or fractured {your/his/her} . . .hip?
- OSQ060 : doc_osteoporosis - Has a doctor ever told {you/SP} that {you/s/he} had osteoporosis, sometimes called thin or brittle bones?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [509]:
# OSQ_2015_2016 = pd.read_csv('./Questionnaire/OSQ_I.csv')

# OSQ_I = OSQ_2015_2016.filter(['SEQN', 'OSQ010A', 'OSQ060'], axis=1)
# OSQ_cols = {'SEQN':'SEQN', 'OSQ010A': 'broken_Iip', 'OSQ060':'doc_osteoporosis'}

# OSQ_I.columns = OSQ_I.columns.map(OSQ_cols)

In [510]:
# OSQ_2015_2016.columns

In [511]:
# OSQ_I.head()

### Physical Activity

variables:
- PAQ650 - Vigorous recreational activities - In a typical week {do you/does SP} do any vigorous-intensity sports, fitness, or recreational activities that cause large increases in breathing or heart rate like running or basketball for at least 10 minutes continuously? 1 yes 2 no
    - PAD200 - Vigorous activity over past 30 days ^^
- PAQ665 - Moderate recreational activities
    - PAD320 - Moderate activity over past 30 days ^^
- PAD680 : min_sedetary - The following question is about sitting at school, at home, getting to and from places, or with friends including time spent sitting at a desk, traveling in a car or bus, reading, playing cards, watching television, or using a computer. Do not include time spent sleeping. How much time {do you/does SP} usually spend sitting on a typical day?
- PAQ706 - Days physically active at least 60 min.
    - 1-7, 77 99
- PAQ710 - Hours watch TV or videos past 30 days
    - 1-5, 8=don’t watch, 77, 99
- PAQ715 - Hours use computer past 30 days (on average/day)
    - “ “
- PAQ722 - Any physical activity in past 7 days

In [512]:
PAQ_2015_2016 = pd.read_csv('./Questionnaire/PAQ_I.csv')

PAQ_I = PAQ_2015_2016.filter(['SEQN', 'PAQ650', 'PAD200', 'PAQ665', 'PAD320',
           'PAD680', 'PAQ706', 'PAQ710','PAQ715','PAQ722'], axis=1)

PAQ_cols = {'SEQN':'SEQN', 'PAQ650':'vigorous_activity', 'PAD200':'vigorous_activity', 'PAQ665':'moderate_activity',
           'PAD3205':'moderate_activity', 'PAD680':'min_sedetary', 'PAQ706':'days_60min_active', 
            'PAQ710':'hours_TV','PAQ715':'hours_computer','PAQ722':'activity_week'}

PAQ_I.columns = PAQ_I.columns.map(PAQ_cols)

In [513]:
PAQ_I.columns

Index(['SEQN', 'vigorous_activity', 'moderate_activity', 'min_sedetary',
       'days_60min_active', 'hours_TV', 'hours_computer', 'activity_week'],
      dtype='object')

In [514]:
PAQ_I.head()

SEQN  vigorous_activity  moderate_activity  min_sedetary  \
0  83732.0                2.0                1.0         480.0   
1  83733.0                2.0                2.0         300.0   
2  83734.0                2.0                2.0         480.0   
3  83735.0                2.0                2.0         480.0   
4  83736.0                2.0                2.0         540.0   

   days_60min_active  hours_TV  hours_computer  activity_week  
0                NaN       5.0    5.397605e-79            NaN  
1                NaN       5.0    5.397605e-79            NaN  
2                NaN       5.0    8.000000e+00            NaN  
3                NaN       3.0    2.000000e+00            NaN  
4                NaN       4.0    5.000000e+00            NaN

### Physical Functioning

PFQ_I
- PFQ049 - Limitations keeping you from working
    - 1,2,7,9
- PFQ059 - Physical, mental, emotional limitations
    - 1,2,7,9
- PFQ061A - Managing money difficulty
    - 1 no difficulty to 4 unable to do, 5 don’t do it, 7, 9

Many other activities I could include

In [515]:
PFQ_2015_2016 = pd.read_csv('./Questionnaire/PFQ_I.csv')

PFQ_I = PFQ_2015_2016.filter(['SEQN', 'PFQ049', 'PFQ059', 'PFQ061A'], axis=1)

PFQ_cols = {'SEQN':'SEQN', 'PFQ049':'limit_work', 'PFQ059':'PME_limit', 'PFQ061A':'manage_$_difficult'}

PFQ_I.columns = PFQ_I.columns.map(PFQ_cols)

In [516]:
PFQ_I.head()

SEQN  limit_work  PME_limit  manage_$_difficult
0  83732.0         2.0        NaN                 1.0
1  83733.0         2.0        2.0                 NaN
2  83734.0         2.0        NaN                 2.0
3  83735.0         2.0        2.0                 NaN
4  83736.0         2.0        NaN                 1.0

### Sleeping Disorders

variables:
- SLQ060 : doc_sleeping_disorder - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/s/he has} a sleep disorder?
- SLD010H - How much sleep do you get (hours)?
    - 2-12, 77, 99

In [517]:
SLQ_2015_2016 = pd.read_csv('./Questionnaire/SLQ_I.csv')

SLQ_I = SLQ_2015_2016.filter(['SEQN', 'SLQ060','SLD010H','SLD012'], axis=1)
SLQ_cols = {'SEQN':'SEQN', 'SLQ060': 'doc_sleeping_disorder','SLD012':'hrs_sleep'}

SLQ_I.columns = SLQ_I.columns.map(SLQ_cols)

In [518]:
SLQ_2015_2016.columns

Index(['SEQN', 'SLQ300', 'SLQ310', 'SLD012', 'SLQ030', 'SLQ040', 'SLQ050',
       'SLQ120'],
      dtype='object')

In [519]:
SLQ_I.head()

SEQN  hrs_sleep
0  83732.0        5.5
1  83733.0        8.0
2  83734.0        7.0
3  83735.0        6.5
4  83736.0        NaN

### Smoking - Cigarette Use (SMQ_I)

variables:
- SMQ020 : smoker - These next questions are about cigarette smoking and other tobacco use. {Have you/Has SP} smoked at least 100 cigarettes in {your/his/her} entire life?

In [520]:
SMQ_2015_2016 = pd.read_csv('./Questionnaire/SMQ_I.csv')

SMQ_I = SMQ_2015_2016.filter(['SEQN', 'SMQ020'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ020': 'smoker'}

SMQ_I.columns = SMQ_I.columns.map(SMQ_cols)

In [521]:
SMQ_I.head()

SEQN  smoker
0  83732.0     1.0
1  83733.0     1.0
2  83734.0     1.0
3  83735.0     2.0
4  83736.0     2.0

In [522]:
SMQRTU_2015_2016 = pd.read_csv('./Questionnaire/SMQRTU_I.csv')

SMQRTU_I = SMQRTU_2015_2016.filter(['SEQN', 'SMQ681'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ681': 'smoked_5days'}

SMQRTU_I.columns = SMQRTU_I.columns.map(SMQ_cols)

In [523]:
SMQRTU_I.head()

SEQN  smoked_5days
0  83732.0           2.0
1  83733.0           1.0
2  83734.0           2.0
3  83735.0           2.0
4  83736.0           1.0

### Sexual Behavior (SXQ_I)

variables:
- SXQ292 : sexual_orientation_M - Do you think of yourself as...1 Heterosexual or straight (attracted to women), 2 Homosexual or gay (attracted to men), 3 Bisexual (attracted to men and women), 4 Something else, 5 Not sure
- SXQ294 : sexual_orientation_F - ""
- SXQ753 : doc_IPV - Has a doctor or other health care professional ever told you that you had human papillomavirus or HPV? 1 yes, 2 no, 7 refuse, 9 don't know
- SXD171 : lifetime_female_partners - In your lifetime, with how many women have you had any kind of sex?
- SXD101 : lifetime_male_partners - In your lifetime, with how many men have you had any kind of sex?


In [524]:
SXQ_2015_2016 = pd.read_csv('./Questionnaire/SXQ_I.csv')

SXQ_I = SXQ_2015_2016.filter(['SEQN', 'SXQ292', 'SXQ294', 'SXD171', 'SXD101', 'SXQ753'], axis=1)
SXQ_cols = {'SEQN':'SEQN', 'SXQ292' : 'sexual_orientation_M', 'SXQ294' : 'sexual_orientation_F', 
           'SXD171' : 'lifetime_female_partners', 'SXD101' : 'lifetime_male_partners', 'SXQ753' : 'doc_HPV'}

SXQ_I.columns = SXQ_I.columns.map(SXQ_cols)


In [525]:
SXQ_2015_2016.columns

Index(['SEQN', 'SXD021', 'SXQ800', 'SXQ803', 'SXQ806', 'SXQ809', 'SXQ700',
       'SXQ703', 'SXQ706', 'SXQ709', 'SXD031', 'SXD171', 'SXD510', 'SXQ824',
       'SXQ827', 'SXD633', 'SXQ636', 'SXQ639', 'SXD642', 'SXQ410', 'SXQ550',
       'SXQ836', 'SXQ841', 'SXQ853', 'SXD621', 'SXQ624', 'SXQ627', 'SXD630',
       'SXQ645', 'SXQ648', 'SXQ610', 'SXQ251', 'SXQ590', 'SXQ600', 'SXD101',
       'SXD450', 'SXQ724', 'SXQ727', 'SXQ130', 'SXQ490', 'SXQ741', 'SXQ753',
       'SXQ260', 'SXQ265', 'SXQ267', 'SXQ270', 'SXQ272', 'SXQ280', 'SXQ295',
       'SXQ296'],
      dtype='object')

In [526]:
SXQ_I.head()

SEQN  lifetime_female_partners  lifetime_male_partners  doc_HPV
0  83732.0                       5.0                     NaN      NaN
1  83733.0                      63.0                     NaN      NaN
2  83735.0                       NaN            5.397605e-79      NaN
3  83736.0                       NaN            1.000000e+01      2.0
4  83741.0                      15.0                     NaN      NaN

notes: may want to concat the # of partners and sexual orientation despite gender

### Taste & Smell

CSQ_I
- CSQ010 - Had problem with smell past 12 months?
    - 1,2,7,9
- CSQ080 - Had problem with taste past 12 months?
    - 1,2,7,9
- CSQ200 - Persistent cold/flu last 12 months?
    - “ “
- CSQ204 - Frequent nasal congestion in past 12 mth

In [527]:
# CSQ_2015_2016 = pd.read_csv('./Questionnaire/CSQ_I.csv')

# CSQ_I = CSQ_2015_2016.filter(['SEQN', 'CSQ010','CSQ080','CSQ200','CSQ204'], axis=1)
# CSQ_cols = {'SEQN':'SEQN', 'CSQ010': 'prob_smell','CSQ080':'prob_tatse',
#             'CSQ200':'persistent_cold','CSQ204':'nasal_congestion'}

# CSQ_I.columns = CSQ_I.columns.map(CSQ_cols)

In [528]:
# CSQ_I.head()

### Alcohol Use (ALQ_I)

Variables: 
- ALQ151 : alcoholic - Was there ever a time or times in {your/SP's} life when {you/he/she} drank {DISPLAY NUMBER} or more drinks of any kind of alcoholic beverage almost every day? 1 yes, 2 no

In [529]:
ALQ_2015_2016 = pd.read_csv('./Questionnaire/ALQ_I.csv')

ALQ_I = ALQ_2015_2016.filter(['SEQN', 'ALQ151'], axis=1)
ALQ_cols = {'SEQN':'SEQN', 'ALQ151': 'alcoholic'}

ALQ_I.columns = ALQ_I.columns.map(ALQ_cols)

In [530]:
ALQ_I.head()

SEQN  alcoholic
0  83732.0        2.0
1  83733.0        1.0
2  83734.0        1.0
3  83735.0        2.0
4  83736.0        2.0

### Weight History

WHQ_I
- WHQ030 - How do you consider your weight
    - 1 - Overweight
    - 2 - Underweight,
    - 3 - About the right weight?
    - 7 - Refused, 9 - Don’t Know
- WHQ070 - Tried to lose weight in past year
    - 1,2,7,9


In [531]:
WHQ_2015_2016 = pd.read_csv('./Questionnaire/WHQ_I.csv')

WHQ_I = WHQ_2015_2016.filter(['SEQN', 'WHQ030','WHQ070'], axis=1)
WHQ_cols = {'SEQN':'SEQN', 'WHQ030':'consider_weight','WHQ070':'tried_lose_weight'}

WHQ_I.columns = WHQ_I.columns.map(WHQ_cols)

In [532]:
WHQ_I.head()

SEQN  consider_weight  tried_lose_weight
0  83732.0              3.0                NaN
1  83733.0              1.0                2.0
2  83734.0              3.0                NaN
3  83735.0              1.0                1.0
4  83736.0              2.0                2.0

## Laboratory Data

### Cholesterol - HDL (HDL_I)

variables:
- LBDHDD : HDL - Direct HDL-Cholesterol (mg/dL), range 10-173

In [533]:
HDL_2015_2016 = pd.read_csv('./Laboratory/HDL_I.csv')

HDL_I = HDL_2015_2016.filter(['SEQN', 'LBDHDD'], axis=1)
HDL_cols = {'SEQN':'SEQN', 'LBDHDD': 'HDL'}

HDL_I.columns = HDL_I.columns.map(HDL_cols)

In [534]:
HDL_I.head()

SEQN   HDL
0  83732.0  46.0
1  83733.0  63.0
2  83734.0  30.0
3  83735.0  61.0
4  83736.0  53.0

### Cholesterol - LDL & Triglycerides (TRIGLY_I)

variables:

- LBDLDL - LBDLDL = ROUND(LBXTC-(LBXHDD+LBXTR/5) for LBXTR less than or equal to 400 mg/dL, and missing for LBXTR greater than 400 mg/dL, range 14-375
- LBXTR : triglyceride - Triglyceride (mg/dL), range 13-4233

In [535]:
TRIGLY_2015_2016 = pd.read_csv('./Laboratory/TRIGLY_I.csv')

TRIGLY_I = TRIGLY_2015_2016.filter(['SEQN', 'LBDLDL', 'LBXTR'], axis=1)
TRIGLY_cols = {'SEQN':'SEQN', 'LBDLDL':'LDL','LBXTR': 'triglyceride'}

TRIGLY_I.columns = TRIGLY_I.columns.map(TRIGLY_cols)

In [536]:
TRIGLY_2015_2016.columns

Index(['SEQN', 'WTSAF2YR', 'LBXTR', 'LBDTRSI', 'LBDLDL', 'LBDLDLSI'], dtype='object')

In [537]:
TRIGLY_I.head()

SEQN    LDL  triglyceride
0  83733.0  173.0         147.0
1  83734.0  145.0         269.0
2  83736.0  142.0          47.0
3  83737.0  103.0          46.0
4  83741.0  102.0          68.0

### Cholesterol - Total (TCHOL_I)

variables:
- LBXTC : tot_cholesterol - Total Cholesterol( mg/dL), range 69-813

In [538]:
TCHOL_2015_2016 = pd.read_csv('./Laboratory/TCHOL_I.csv')

TCHOL_I = TCHOL_2015_2016.filter(['SEQN', 'LBXTC'], axis=1)
TCHOL_cols = {'SEQN':'SEQN', 'LBXTC': 'tot_cholesterol'}

TCHOL_I.columns = TCHOL_I.columns.map(TCHOL_cols)

In [539]:
TCHOL_I.head()

SEQN  tot_cholesterol
0  83732.0            173.0
1  83733.0            265.0
2  83734.0            229.0
3  83735.0            174.0
4  83736.0            204.0

### Glycohemoglobin (GHB_I)

variables:
- LBXGH - Glycohemoglobin (%), range 3.5-17.5

In [540]:
GHB_2015_2016 = pd.read_csv('./Laboratory/GHB_I.csv')

GHB_I = GHB_2015_2016.filter(['SEQN', 'LBXGH'], axis=1)
GHB_cols = {'SEQN':'SEQN', 'LBXGH': 'glycohemoglobin'}

GHB_I.columns = GHB_I.columns.map(GHB_cols)

In [541]:
GHB_I.head()

SEQN  glycohemoglobin
0  83732.0              7.0
1  83733.0              5.5
2  83734.0              5.8
3  83735.0              5.6
4  83736.0              5.6

### Herpes Simplex Virus Type-1 & Type-2 (HSV_I)

variables:
- LBXHE2 : herpes_2 - Herpes Simplex Virus Type 2

In [542]:
HSV_2015_2016 = pd.read_csv('./Laboratory/HSV_I.csv')

HSV_I = HSV_2015_2016.filter(['SEQN', 'LBXHE2'], axis=1)
HSV_cols = {'SEQN':'SEQN', 'LBXHE2': 'herpes_2'}

HSV_I.columns = HSV_I.columns.map(HSV_cols)

In [543]:
HSV_I.head(2)

SEQN  herpes_2
0  83736.0       1.0
1  83741.0       2.0

### HIV Antibody Test (HIV_I)

variables:
- LBDHI : HIV - HIV antibody test result, 1 pos, 2, neg, 3 indeterminant

In [544]:
HIV_2015_2016 = pd.read_csv('./Laboratory/HIV_I.csv')

HIV_I = HIV_2015_2016.filter(['SEQN', 'LBXHIVC'], axis=1)
HIV_cols = {'SEQN':'SEQN', 'LBXHIVC': 'HIV'}

HIV_I.columns = HIV_I.columns.map(HIV_cols)

In [545]:
HIV_I.head(2)

SEQN  HIV
0  83733.0  2.0
1  83735.0  2.0

### High-Sensitivity C-Reactive Protein (hs-CRP) (HSCRP_I) - NOT IN 2013-2014

variables:
- LBXHSCRP : CRP - HS C-Reactive Protein (mg/L), values 0-188, missing values are mostly below detection limit --> can be made to 0

In [546]:
HSCRP_2015_2016 = pd.read_csv('./Laboratory/HSCRP_I.csv')

HSCRP_I = HSCRP_2015_2016.filter(['SEQN', 'LBXHSCRP'], axis=1)
HSCRP_cols = {'SEQN':'SEQN', 'LBXHSCRP': 'CRP'}

HSCRP_I.columns = HSCRP_I.columns.map(HSCRP_cols)

In [547]:
HSCRP_I.head()

SEQN  CRP
0  83732.0  0.6
1  83733.0  1.4
2  83734.0  0.6
3  83735.0  9.0
4  83736.0  0.5

### Serums

CUSEZN_I
- LBXSSE - Serum Selenium (ug/L) - cont range, missing
- LBXSZN - Serum Zinc (ug/dL) - continuous range, missing
- URXUCR - Urinary creatinine (mg/dL) - continuous range, missing

In [548]:
CUSEZN_2015_2016 = pd.read_csv('./Laboratory/CUSEZN_I.csv')

CUSEZN_I = CUSEZN_2015_2016.filter(['SEQN', 'LBXSSE','LBXSZN','URXUCR'], axis=1)
CUSEZN_cols = {'SEQN':'SEQN', 'LBXSSE': 'serum_selenium', 'LBXSZN':'serum_zinc','URXUCR':'urine_creatinine'}

CUSEZN_I.columns = CUSEZN_I.columns.map(CUSEZN_cols)

In [549]:
CUSEZN_I.head()

SEQN  serum_selenium  serum_zinc
0  83732.0           141.1        88.5
1  83733.0           130.4       100.6
2  83734.0           126.4        98.7
3  83738.0           110.4        61.1
4  83741.0           119.9        86.2

### Folate

FOLATE_I
- LBDRFO - RBC folate (ng/mL) -  continuous range, missing

FOLFMS_I
- LBDFOT - Serum total folate (ng/mL) -  continuous range, missing

In [550]:
FOLATE_2015_2016 = pd.read_csv('./Laboratory/FOLATE_I.csv')
FOLFMS_2015_2016 = pd.read_csv('./Laboratory/FOLFMS_I.csv')

FOLATE_I = FOLATE_2015_2016.filter(['SEQN', 'LBDRFO'], axis=1)
FOLATE_cols = {'SEQN':'SEQN', 'LBDRFO':'rbc_folate'}
FOLFMS_I = FOLFMS_2015_2016.filter(['SEQN', 'LBDFOT'], axis=1)
FOLFMS_cols = {'SEQN':'SEQN', 'LBDFOT':'serum_folate'}

FOLATE_I.columns = FOLATE_I.columns.map(FOLATE_cols)
FOLFMS_I.columns = FOLFMS_I.columns.map(FOLFMS_cols)

In [551]:
FOLFMS_I.head()

SEQN  serum_folate
0  83732.0          37.1
1  83733.0          11.6
2  83734.0          29.3
3  83735.0           8.7
4  83736.0          12.1

In [552]:
FOLATE_I.head()

SEQN  rbc_folate
0  83732.0       812.0
1  83733.0       413.0
2  83734.0      1080.0
3  83735.0       374.0
4  83736.0       196.0

### Fasting Glucose

GLU_I
- LBXGLU - Fasting Glucose (mg/dL) - 51 to 421, missing

In [553]:
GLU_2015_2016 = pd.read_csv('./Laboratory/GLU_I.csv')

GLU_I = GLU_2015_2016.filter(['SEQN', 'LBXGLU'], axis=1)
GLU_cols = {'SEQN':'SEQN', 'LBXGLU': 'fasting_glucose'}

GLU_I.columns = GLU_I.columns.map(GLU_cols)

In [554]:
GLU_I.head()

SEQN  fasting_glucose
0  83733.0            101.0
1  83734.0             84.0
2  83736.0             84.0
3  83737.0            107.0
4  83741.0             95.0

### Vitamins B12 and D

VITB12_I
- LBDB12 - Vitamin B12(pg/mL) - range, missing

VID_I
- LBXVIDMS - 25OHD2+25OHD3 (nmol/L) - range, missing (vitamin D)

In [555]:
# VITB12_2015_2016 = pd.read_csv('./Laboratory/VITB12_I.csv')

# VITB12_I = VITB12_2015_2016.filter(['SEQN', 'LBDB12'], axis=1)
# VITB12_cols = {'SEQN':'SEQN', 'LBDB12': 'lab_VitB12'}

# VITB12_I.columns = VITB12_I.columns.map(VITB12_cols)


# VID_2015_2016 = pd.read_csv('./Laboratory/VID_I.csv')

# VID_I = VID_2015_2016.filter(['SEQN', 'LBXVIDMS'], axis=1)
# VID_cols = {'SEQN':'SEQN', 'LBXVIDMS': 'lab_VitD'}

# VID_I.columns = VID_I.columns.map(VID_cols)


In [556]:
#VITB12_I.head()

In [557]:
#VID_I.head()

## Examination Data

### Blood Pressure (BPX_I)

variables:
- BPXSY1 : systolic_BP - Systolic: Blood pres (1st rdg) mm Hg
- BPXDI1 : diastolic_BP - Diastolic: Blood pres (1st rdg) mm Hg
- BPXPLS - 60 sec. pulse (30 sec. pulse * 2) - 0 to 180, missing

In [558]:
BPX_2015_2016 = pd.read_csv('./Examination/BPX_I.csv')

BPX_I = BPX_2015_2016.filter(['SEQN', 'BPXSY1', 'BPXDI1','BPXPLS'], axis=1)
BPX_cols = {'SEQN':'SEQN', 'BPXSY1': 'systolic_BP', 'BPXDI1' : 'diastolic_BP','BPXPLS':'pulse'}

BPX_I.columns = BPX_I.columns.map(BPX_cols)

In [559]:
BPX_I.head(2)

SEQN  systolic_BP  diastolic_BP  pulse
0  83732.0        128.0          70.0   76.0
1  83733.0        146.0          88.0   72.0

### Body Measures (BMX_I)

variables:
- BMXBMI : BMI - Body Mass Index (kg/m**2), range 12.1 to 82.9
- BMXWAIST : waist_C - Waist Circumference (cm), 40.2 to 177.9

In [560]:
BMX_2015_2016 = pd.read_csv('./Examination/BMX_I.csv')

BMX_I = BMX_2015_2016.filter(['SEQN', 'BMXBMI', 'BMXWAIST'], axis=1)
BMX_cols = {'SEQN':'SEQN', 'BMXBMI': 'BMI', 'BMXWAIST' : 'waist_C'}

BMX_I.columns = BMX_I.columns.map(BMX_cols)

In [561]:
BMX_I.head(2)

SEQN   BMI  waist_C
0  83732.0  27.8    101.1
1  83733.0  30.8    107.9

### Dual-Energy X-ray Absorptiometry - Whole Body (DXX_I)

variables:
- DXDTOBMD : bone_mineral_density - Total Bone Mineral Density (g/cm^2), range 0.559 to 1.548

In [562]:
DXX_2015_2016 = pd.read_csv('./Examination/DXX_I.csv')

DXX_I = DXX_2015_2016.filter(['SEQN', 'DXDTOBMD'], axis=1)
DXX_cols = {'SEQN':'SEQN', 'DXDTOBMD': 'bone_mineral_density'}

DXX_I.columns = DXX_I.columns.map(DXX_cols)

In [563]:
DXX_I.head(2)

SEQN  bone_mineral_density
0  83733.0                 1.082
1  83735.0                 1.025

### Oral Health

OHXREF_I
- OHAREC - Overall recommendation for care
    - 1 - See a dentist immediately
    - 2 - See a dentist within the next 2 weeks	
    - 3 - See a dentist at your earliest convenience
    - 4 - Continue your regular routine care
    - Missing


In [564]:
OHXREF_2015_2016 = pd.read_csv('./Examination/OHXREF_I.csv')

OHXREF_I = OHXREF_2015_2016.filter(['SEQN', 'OHAREC'], axis=1)
OHXREF_cols = {'SEQN':'SEQN', 'OHAREC': 'oral_rec'}

OHXREF_I.columns = OHXREF_I.columns.map(OHXREF_cols)

In [565]:
OHXREF_I.head()

SEQN  oral_rec
0  83732.0       4.0
1  83733.0       4.0
2  83734.0       4.0
3  83735.0       4.0
4  83736.0       3.0

# Merging dataframes

In [566]:
from functools import reduce

# list of df
# OSQ_I, VITB12_I, VID_I, CSQ_I,
data_frames = [DPQ_I, DEMO_I, DIET_I, DIQ_I, DBQ_I, DLQ_I, DUQ_I, HUQ_I, HIQ_I, 
               HOQ_I, ECQ_I, FSQ_I, INQ_I, MCQ_I, KIQ_U_I, OCQ_I, OHQ_I, 
               PAQ_I, PFQ_I, SLQ_I, SMQ_I, SMQRTU_I, SXQ_I, ALQ_I, WHQ_I, 
               HDL_I, TRIGLY_I, TCHOL_I, GHB_I, HSV_I, HIV_I, CUSEZN_I, FOLFMS_I, 
               FOLATE_I, GLU_I, BPX_I, BMX_I, DXX_I, OHXREF_I]

df_2015_2016 = reduce(lambda  left,right: pd.merge(left,right,on=['SEQN'],
                                            how='outer'), data_frames)

In [567]:
df_2015_2016.depressed

0       0.0
1       0.0
2       0.0
3       1.0
4       0.0
       ... 
9966    NaN
9967    NaN
9968    NaN
9969    NaN
9970    NaN
Name: depressed, Length: 9971, dtype: float64

In [568]:
df_2015_2016.dropna(subset=['depressed'], inplace=True)

In [569]:
df_2015_2016.shape

(5161, 107)

In [570]:
df_2015_2016.columns

Index(['SEQN', 'depressed', 'race_ethnicity', 'edu_level', '#_ppl_household',
       'age', 'gender', 'annual_HI', 'marital_status', 'caffeine',
       ...
       'serum_folate', 'rbc_folate', 'fasting_glucose', 'systolic_BP',
       'diastolic_BP', 'pulse', 'BMI', 'waist_C', 'bone_mineral_density',
       'oral_rec'],
      dtype='object', length=107)

# Export 2015-2016 data

In [571]:
df_2015_2016.to_csv('/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/df_2015_2016_ext.csv', index=False)